google/cloud/spanner/retry_policy.h

```diff
@@ -59,33 +59,267 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_END
+      spanner_internal::SafeTransactionRerun>
+      impl_;
+};
+
+/**
+ * A transaction rerun policy based on elapsed time.
+ *
+ * This policy stops retrying if:
+ * - An RPC returns a non-transient error.
+ * - The elapsed time in the retry loop exceeds a prescribed duration.
+ *
+ * The following status codes are treated as transient errors:
+ * - [`kAborted`](@ref google::cloud::StatusCode)
+ * - [`kNotFound`](@ref google::cloud::StatusCode) when the error message
+ *   indicates the missing (or expired) resource is the spanner session
+ *   associated with the transaction.
+ */
+class LimitedTimeTransactionRerunPolicy : public TransactionRerunPolicy {
+ public:
+  /**
+   * Constructor given a `std::chrono::duration<>` object.
+   *
+   * @tparam DurationRep a placeholder to match the `Rep` tparam for
+   *     @p maximum_duration's type. The semantics of this template parameter
+   *     are documented in `std::chrono::duration<>`. In brief, the underlying
+   *     arithmetic type used to store the number of ticks. For our purposes it
+   *     is simply a formal parameter.
+   * @tparam DurationPeriod a placeholder to match the `Period` tparam for
+   *     @p maximum_duration's type. The semantics of this template parameter
+   *     are documented in `std::chrono::duration<>`. In brief, the length of
+   *     the tick in seconds, expressed as a `std::ratio<>`. For our purposes it
+   *     is simply a formal parameter.
+   * @param maximum_duration the maximum time allowed before the policy expires,
+   *     while the application can express this time in any units they desire,
+   *     the class truncates to milliseconds.
+   *
+   * @see https://en.cppreference.com/w/cpp/chrono/duration for more details
+   *     about `std::chrono::duration`.
+   */
+  template <typename DurationRep, typename DurationPeriod>
+  explicit LimitedTimeTransactionRerunPolicy(
+      std::chrono::duration<DurationRep, DurationPeriod> maximum_duration)
+      : impl_(maximum_duration) {}
+
+  LimitedTimeTransactionRerunPolicy(
+      LimitedTimeTransactionRerunPolicy&& rhs) noexcept
+      : LimitedTimeTransactionRerunPolicy(rhs.maximum_duration()) {}
+  LimitedTimeTransactionRerunPolicy(
+      LimitedTimeTransactionRerunPolicy const& rhs) noexcept
+      : LimitedTimeTransactionRerunPolicy(rhs.maximum_duration()) {}
+
+  std::chrono::milliseconds maximum_duration() const {
+    return impl_.maximum_duration();
+  }
+
+  bool OnFailure(Status const& s) override { return impl_.OnFailure(s); }
+  bool IsExhausted() const override { return impl_.IsExhausted(); }
+  bool IsPermanentFailure(Status const& s) const override {
+    return impl_.IsPermanentFailure(s);
+  }
+  std::unique_ptr<TransactionRerunPolicy> clone() const override {
+    return std::make_unique<LimitedTimeTransactionRerunPolicy>(
+        impl_.maximum_duration());
+  }
+
+  // Not very useful, but needed for backwards compatibility.
```

Comment:
Nit: Use the same "This is provided only for backwards compatibility" text as the other ones?

Reply:
Fixed, and added one missing typedef.